tests/e2e/vsphere_volume_expansion.go

```diff
@@ -412,6 +421,470 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+		ginkgo.By("Invoking Test for Volume Expansion")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sharedNFSdatastoreURL := os.Getenv(envSharedNFSDatastoreURL)
+		if sharedNFSdatastoreURL == "" {
+			ginkgo.Skip("env SHARED_NFS_DATASTORE_URL is missing")
+		}
+
+		ginkgo.By("Create StorageClass on shared NFS datastore with allowVolumeExpansion set to true")
+		volHandle, pvclaim, pv, storageclass := createSCwithVolumeExpansionTrueAndDynamicPVC(f, client, sharedNFSdatastoreURL, namespace)
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Create POD using the above PVC")
+		pod := createPODandVerifyVolumeMount(f, client, namespace, pvclaim, volHandle)
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		}()
+
+		ginkgo.By("Increase PVC size and verify Volume resize")
+		increaseSizeOfPvcAttachedToPod(f, client, namespace, pvclaim, pod)
+
+	})
+
+	/*
+		Verify online volume expansion on Shared VMFS datastore
+		1. Create StorageClass with allowVolumeExpansion set to true on a shared VMFS datastore.
+		2. Create PVC which uses the StorageClass created in step 1.
+		3. Wait for PV to be provisioned.
+		4. Wait for PVC's status to become Bound and note down the size
+		5. Create a POD using the above created PVC
+		6. Modify PVC's size to trigger online volume expansion
+		7. verify the PVC status will change to "FilesystemResizePending". Wait till the status is removed
+		8. Verify the resized PVC by doing CNS query
+		9. Make sure data is intact on the PV mounted on the pod
+		10.  Make sure file system has increased
+	*/
+	ginkgo.It("[csi-block-vanilla] Verify online volume expansion on shared VMFS datastore", func() {
+		ginkgo.By("Invoking Test for Volume Expansion")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sharedVMFSdatastoreURL := os.Getenv(envSharedVMFSDatastoreURL)
+		if sharedVMFSdatastoreURL == "" {
+			ginkgo.Skip("env SHARED_VMFS_DATASTORE_URL is missing")
```

Comment:
We shouldn't just skip tests if ENV is not set. What if someone has VMFS datastore in the testbed and forget to set the ENV. We should fail, right ?

Reply:
We are giving the details in the doc saying what all the things need to set before running the tests . It is documented here vsphere-csi-driver/tests/e2e/docs/vanilla_cluster_setup.md

I agree that message what i am setting may not be sufficient . I have updated the message